csharp/src/Drivers/BigQuery/BigQueryConnection.cs

```diff
@@ -0,0 +1,954 @@
+                }
+            }
+
+            List<IArrowArray> dataArrays = new List<IArrowArray>
+            {
+                tableNameBuilder.Build(),
+                tableTypeBuilder.Build(),
+                CreateNestedListArray(tableColumnsValues, new StructType(StandardSchemas.ColumnSchema)),
+                CreateNestedListArray(tableConstraintsValues, new StructType(StandardSchemas.ConstraintSchema))
+            };
+
+            return new StructArray(
+                new StructType(StandardSchemas.TableSchema),
+                length,
+                dataArrays,
+                nullBitmapBuffer.Build());
+        }
+
+        private StructArray GetColumnSchema(
+            string catalog,
+            string dbSchema,
+            string table,
+            string columnNamePattern)
+        {
+            StringArray.Builder columnNameBuilder = new StringArray.Builder();
+            Int32Array.Builder ordinalPositionBuilder = new Int32Array.Builder();
+            StringArray.Builder remarksBuilder = new StringArray.Builder();
+            Int16Array.Builder xdbcDataTypeBuilder = new Int16Array.Builder();
+            StringArray.Builder xdbcTypeNameBuilder = new StringArray.Builder();
+            Int32Array.Builder xdbcColumnSizeBuilder = new Int32Array.Builder();
+            Int16Array.Builder xdbcDecimalDigitsBuilder = new Int16Array.Builder();
+            Int16Array.Builder xdbcNumPrecRadixBuilder = new Int16Array.Builder();
+            Int16Array.Builder xdbcNullableBuilder = new Int16Array.Builder();
+            StringArray.Builder xdbcColumnDefBuilder = new StringArray.Builder();
+            Int16Array.Builder xdbcSqlDataTypeBuilder = new Int16Array.Builder();
+            Int16Array.Builder xdbcDatetimeSubBuilder = new Int16Array.Builder();
+            Int32Array.Builder xdbcCharOctetLengthBuilder = new Int32Array.Builder();
+            StringArray.Builder xdbcIsNullableBuilder = new StringArray.Builder();
+            StringArray.Builder xdbcScopeCatalogBuilder = new StringArray.Builder();
+            StringArray.Builder xdbcScopeSchemaBuilder = new StringArray.Builder();
+            StringArray.Builder xdbcScopeTableBuilder = new StringArray.Builder();
+            BooleanArray.Builder xdbcIsAutoincrementBuilder = new BooleanArray.Builder();
+            BooleanArray.Builder xdbcIsGeneratedcolumnBuilder = new BooleanArray.Builder();
+            ArrowBuffer.BitmapBuilder nullBitmapBuffer = new ArrowBuffer.BitmapBuilder();
+            int length = 0;
+
+            string query = string.Format("SELECT * FROM `{0}`.`{1}`.INFORMATION_SCHEMA.COLUMNS WHERE table_name = '{2}'",
+                catalog, dbSchema, table);
+
+            if (columnNamePattern != null)
+            {
+                query = string.Concat(query, string.Format("AND column_name LIKE '{0}'", columnNamePattern));
```

Comment:
columnNamePattern comes from the caller, making this susceptible to SQL injection

Reply:
fixed in latest push